lib/server.js

```diff
@@ -0,0 +1,205 @@
+import http from 'node:http'
+import net from 'node:net'
+import {EventEmitter} from 'node:events'
+import crypto from 'node:crypto'
+import url from 'node:url'
+
+import {Headers} from 'undici'
+
+import {SafeHttpClient, HttpError} from './safe-http-client.js'
+import {
+  DEFAULT_HEADERS,
+  REQUEST_HEADERS,
+  RESPONSE_HEADERS
+} from './constants.js'
+
+/**
+ * @typedef {Object} Config
+ * @property {string} HMACKey
+ * @property {string} serverName Server name used in Headers and Via checks
+ * @property {number|undefined} [maxSize] The max size in bytes per resource to download. Sends 500 if exceeded.
+ */
```

Comment:
```suggestion

/**
 * @typedef Config
 *   Configuration.
 * @property {string} HMACKey
 *   Secret.
 * @property {string} serverName
 *   Name of server; used in `Headers` and `Via`.
 * @property {number | undefined} [maxSize]
 *   Max size in bytes per resource to download (default: `undefined`);
 *   If passed, sends a `404` if exceeded.
 */
```

More docs.

Q1: I prefer the name `Options` for such types.
Q2: how about `secret`, just like `rehype-github-images`, for the shared secret? Otherwise, I prefer `hmacKey`
Q3: I don’t think `serverName` is actually “checked” (current text?), it’s sent along though?
Q4: `maxSize` currently seems to send a 404?
Q5: should we set a default `maxSize`?

Reply:
> Q1: I prefer the name `Options` for such types.

👍 

> Q2: how about `secret`, just like `rehype-github-images`, for the shared secret? Otherwise, I prefer `hmacKey`

👍 

> Q3: I don’t think `serverName` is actually “checked” (current text?), it’s sent along though?

Checked for what? It being a string you mean?

> Q4: `maxSize` currently seems to send a 404?

Yes thinking to change it to `413` (Content Too Large).

> Q5: should we set a default `maxSize`?

Probably a good idea yes. Here is what GPT thinks the default should be: https://chat.openai.com/share/188485eb-b6ee-4957-a96a-c112c1582544